include/ensmallen_bits/cmaes/cmaes_impl.hpp

```diff
@@ -128,7 +128,20 @@ double CMAES<SelectionPolicyType>::Optimize(
     const size_t idx0 = (i - 1) % 2;
     const size_t idx1 = i % 2;
 
-    const arma::mat covLower = arma::chol(C.slice(idx0), "lower");
+    arma::mat covLower; 
+
+    try
+    {
+      covLower = arma::chol(C.slice(idx0), "lower");
+    }
+    catch (const std::runtime_error& err)
+    {
+      // If the covariance matrix isn't positive definite, add a small amount
+      // to the diagonal and try again
+      arma::mat I(iterate.n_elem, iterate.n_elem, arma::fill::eye);
+      C.slice(idx0) += I*(1e-16);
+      covLower = arma::chol(C.slice(idx0), "lower");
+    }
```

Comment:
Actually you can simplify this a bit...

```
C.slice(idx0).diag() += 1e-16;
```

Also, if `1e-16` is generally sufficient I'm okay with that, but it's also possible to loop iteratively until the Cholesky decomposition succeeds.  You could use the version `chol(C.slice(idx0), covLower, "lower")` which returns a `bool`, and loop until it's true.

Reply:
Here is a simplification. :)

```
while (!arma::chol(covLower, C.slice(idx0), "lower"))
  C.slice(idx0).diag() += 1e-16;
```

Reply:
Yup, far simpler, thank you for the suggestion